packages/compass-app-stores/src/provider.tsx

```diff
@@ -93,6 +94,32 @@ export const mongoDBInstanceLocator = createServiceLocator(
   'mongoDBInstanceLocator'
 );
 
+/**
+ * Hook to check if there are any connected non-genuine MongoDB connections.
+ * @return true if any connected connection is to a non-genuine MongoDB server.
+ */
+export function useHasNonGenuineConnections(): boolean {
```

Comment:
I'm suggesting to add a re-usable hook in this package as it needs both connection state and instance state.

It seems to me, this package is the integration layer that already bridges those two concepts - but there might be a better place for it?

Reply:
Yes, more or less, this is a logical place to have something like this, although you'll need to reshuffle things a bit to make it fully work:

- You can't use locators directly (see my other comment), so you'd need to reshuffle the code here a bit and create the context accessing hooks first separately from locator code (like literally just move the functions defined inside `createServiceLocator` out for now is good enough)
- Values on MongoDBInstance are dynamic and your code right now is not accounting for that, usually with ampersand models we set up a change listener for the value to track its state updates after initial read. We can assume this particular logic is not needed here and can be skipped, but let's document this at least